main/util/src/mill/util/CoursierSupport.scala

```diff
@@ -164,6 +167,47 @@ trait CoursierSupport {
     (deps0, res.getOrThrow)
   }
 
+  def jvmIndex(
+      ctx: Option[mill.api.Ctx.Log] = None,
+      coursierCacheCustomizer: Option[FileCache[Task] => FileCache[Task]] = None
+  ): Result[JvmIndex] = {
+    val coursierCache0 = coursierCache(ctx, coursierCacheCustomizer)
+    JvmIndex.load().unsafeRun()(coursierCache0.ec)
+  }
+
+  /**
+   * Resolve java home using Coursier.
+   *
+   * The id string has format "$DISTRIBUTION:$VERSION". e.g. graalvm-community:23.0.0
+   */
+  def resolveJavaHome(
+      id: String,
+      ctx: Option[mill.api.Ctx.Log] = None,
+      coursierCacheCustomizer: Option[FileCache[Task] => FileCache[Task]] = None
+  ): Result[os.Path] = {
+    val coursierCache0 = coursierCache(ctx, coursierCacheCustomizer)
+    val jvmCache = JvmCache()
+      .withArchiveCache(
+        ArchiveCache().withCache(coursierCache0)
+      )
+      .withIndex(JvmIndex.load(
+        cache = coursierCache0, // the coursier.cache.Cache instance to use
+        repositories = Resolve().repositories, // repositories to use
+        indexChannel = JvmChannel.central(), // use new indices published to Maven Central
```

Comment:
Not much, maybe I should add one. The new indices live under [this organization](https://repo1.maven.org/maven2/io/get-coursier/jvm/indices)

Reply:
Yes please do. Especially for the questions I asked below, people will want to know how this stuff works in case they need to debug it when it goes wrong